crates/http/src/server.rs

```diff
@@ -4,25 +4,30 @@ use std::net::SocketAddr;
 use crate::error::Error;
 use crate::service::{HttpService, HttpServiceFactory};
 
-pub mod builder;
+// pub mod builder;
 
 /// The HTTP server.
 ///
 /// This struct is the main entry point for creating and running an HTTP server.
 ///
 /// Create a new server using the [`ServerBuilder`](builder::ServerBuilder) struct.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, bon::Builder)]
+#[builder(state_mod(vis = "pub(crate)"))]
 #[allow(dead_code)]
-pub struct HttpServer<S> {
+pub struct HttpServer<F> {
+    #[builder(default)]
     ctx: scuffle_context::Context,
-    service_factory: S,
+    service_factory: F,
     bind: SocketAddr,
+    #[builder(default = true)]
     #[cfg(feature = "http1")]
     #[cfg_attr(docsrs, doc(cfg(feature = "http1")))]
     enable_http1: bool,
+    #[builder(default = true)]
     #[cfg(feature = "http2")]
     #[cfg_attr(docsrs, doc(cfg(feature = "http2")))]
     enable_http2: bool,
+    #[builder(default = false)]
     #[cfg(feature = "http3")]
     #[cfg_attr(docsrs, doc(cfg(feature = "http3")))]
     enable_http3: bool,
```

Comment:
so we know that http3 cannot be enabled without tls enabled. Therefore we should do something like
```rs
pub struct HttpServer<F> {
    // ...
    #[builder(setters(vis = "", name = enable_http3_internal))]
    enable_http3: bool,
    // ...
}
```

```rs
impl<F, S: http_server_builder::State> HttpServerBuilder<F, S> {
    pub fn enable_http3(
        self,
    ) -> HttpServerBuilder<http_server_builder::SetEnableHttp3<F, S>>
    where
        S::RustlsConfig: http_server_builder::IsSet,
    {
        self.enable_http3_internal()
    }
```